src/proto/batch/mod.rs

```diff
@@ -0,0 +1,338 @@
+use std::io::{Read, Write};
+
+use chrono::{DateTime, Utc};
+use derive_builder::Builder;
+
+use crate::{Error, Job, Producer};
+
+mod cmd;
+
+pub use cmd::{CommitBatch, GetBatchStatus, OpenBatch};
+
+/// Batch of jobs.
+///
+/// Faktory guarantees a callback (`success` and/or `failure`) will be triggered after the execution
+/// of all the jobs belonging to the same batch has finished (successfully or with errors accordingly).
+/// The 'complete' callback will always be queued first.
+///
+/// Batches can be nested. They can also be re-opened, but - once a batch is committed - only those jobs
+/// that belong to this batch can re-open it.
+///
+/// An empty batch can be committed just fine. That will make Faktory immediately fire a callback (i.e. put
+/// the job specified in `complete` and/or the one specified in `success` onto the queues).
+///
+/// If you open a batch, but - for some reason - do not commit it within _30 minutes_, it will simply expire
+/// on the Faktory server (which means no callbackes will be fired).
+///
+/// Here is how you can create a simple batch:
+/// ```no_run
+/// # use faktory::Error;
+/// use faktory::{Producer, Job, Batch};
+///
+/// let mut prod = Producer::connect(None)?;
+/// let job1 = Job::builder("image").build();
+/// let job2 = Job::builder("image").build();
+/// let job_cb = Job::builder("clean_up").build();
+///
+/// let batch = Batch::builder("Image resizing workload".to_string()).with_complete_callback(job_cb);
```

Comment:
The batch and job names here and below feel a little random. Maybe use something like "First batch" or "Outer batch"?

Reply:
Now using `Parent batch description` and `Child batch description` for batch descriptions, and also simply `job_type` and `another_job_type`. But yes, I wanted it to be 'real-life' examples, maybe not the best choices for the docs.

Reply:
I like real-life examples, but I think they then belong in an actual example (say, in `examples/`) with more context + end-to-end code. For doc code for individual methods (without that context), it just reads kind of strange.